src/multi_vector_simulator/C0_data_processing.py

```diff
@@ -1776,22 +1777,30 @@ def process_maximum_cap_constraint(dict_values, group, asset, subasset=None):
     else:
         asset_dict = dict_values[group][asset][subasset]
 
-    # Check if a maximumCap is defined
+    # include the maximumAddCap parameter to the asset dictionary
+    asset_dict.update({MAXIMUM_ADD_CAP: {VALUE: None}})
+
+    # check if a maximumCap is defined
     if MAXIMUM_CAP not in asset_dict:
         asset_dict.update({MAXIMUM_CAP: {VALUE: None}})
     else:
         if asset_dict[MAXIMUM_CAP][VALUE] is not None:
-            # set maximumCap to None if it is smaller than installedCap
-            if asset_dict[MAXIMUM_CAP][VALUE] < asset_dict[INSTALLED_CAP][VALUE]:
+            # maximum additional capacity = maximum total capacity - installed capacity
+            max_add_cap = (
+                asset_dict[MAXIMUM_CAP][VALUE] - asset_dict[INSTALLED_CAP][VALUE]
+            )
+            # include the maximumAddCap parameter to the asset dictionary
+            asset_dict.update({MAXIMUM_ADD_CAP: {VALUE: max_add_cap}})
+            # raise error if maximumCap is smaller than installedCap and is not set to zero
+            if (
+                asset_dict[MAXIMUM_CAP][VALUE] < asset_dict[INSTALLED_CAP][VALUE]
+                and asset_dict[MAXIMUM_CAP][VALUE] != 0
+            ):
                 message = (
-                    f"The stated maximumCap in {group} {asset} is smaller than the "
+                    f"The stated total maximumCap in {group} {asset} is smaller than the "
                     f"installedCap ({asset_dict[MAXIMUM_CAP][VALUE]}/{asset_dict[INSTALLED_CAP][VALUE]}). Please enter a greater maximumCap."
-                    "For this simulation, the maximumCap will be "
-                    "disregarded and not be used in the simulation"
                 )
-                warnings.warn(UserWarning(message))
-                logging.warning(message)
-                asset_dict[MAXIMUM_CAP][VALUE] = None
+                raise ValueError(message)
```

Comment:
See disussion above: I am not sure if we should use `ValueError`or a `logging.Error()` here.

I think the if-statement should be placed first, and then the `asset_dict.update({MAXIMIUM_ADD_CAP: {VALUE: max_add_cap})`, as otherwise you may place a negative value in there. Later changes might keep that negative value. 

If using a `logging.error` message is the "correct" way for the EPA, then you could do the check, display error message, state that no capacities can be added, and set `max_add_cap` to 0.

Briefly I considered whether we should change `MAXIMUM_CAP=InstalledCap` or `OPTIMIZE_CAP=False`, but I think this is not the way to go as then we would ourselves change the input values.

Reply:
> If using a `logging.error` message is the "correct" way for the EPA, then you could do the check, display error message, state that no capacities can be added, and set `max_add_cap` to 0.

This is not the "correct" way, the correct way is to raise a ValueError